test/jdk/java/awt/PopupMenu/PopupMenuLocation.java

```diff
@@ -106,22 +119,37 @@ private void show(MouseEvent e) {
     }
 
     private static void openPopup(final Frame frame) throws Exception {
-        Robot robot = new Robot();
-        robot.setAutoDelay(200);
         robot.waitForIdle();
-        Point pt = frame.getLocationOnScreen();
-        robot.mouseMove(pt.x + frame.getWidth() / 2, pt.y + 50);
+        final AtomicReference<Point> pointObj = new AtomicReference<>();
+        SwingUtilities.invokeAndWait(() -> pointObj.set(frame.getLocationOnScreen()));
+        Point pt = pointObj.get();
+        int x = pt.x + frame.getWidth() / 2;
+        int y = pt.y + 50;
+        robot.mouseMove(x, y);
         robot.mousePress(InputEvent.BUTTON3_DOWN_MASK);
         robot.mouseRelease(InputEvent.BUTTON3_DOWN_MASK);
-        int x = pt.x + frame.getWidth() / 2;
-        int y = pt.y + 130;
+         y = y+50;
         robot.mouseMove(x, y);
         robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
         robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
         robot.waitForIdle();
         if (!action) {
-            throw new RuntimeException();
+            captureScreen();
+            throw new RuntimeException(
+                    "Failed, Not received the PopupMenu ActionEvent yet");
         }
         action = false;
     }
+
+    private static void captureScreen() {
+        Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize();
+        try {
+            ImageIO.write(robot.createScreenCapture(
+                    new Rectangle(0, 0, screenSize.width, screenSize.height)),
```

Comment:
This may capture the default screen while the frame is currently visible on another one.

Reply:
But, is there any way we can capture the current screen where the frame is displayed?

Reply:
You can pass coordinates/bounds of that screen to the createScreenCapture.